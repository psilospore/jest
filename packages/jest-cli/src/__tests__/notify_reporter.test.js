/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

'use strict';

import TestScheduler from '../test_scheduler';
import NotifyReporter from '../reporters/notify_reporter';
import type {TestSchedulerContext} from '../test_scheduler';

const SUCCESS_ICON_PATH = '/assets/jest_logo.png';
const FAILURE_ICON_PATH = 'TODO';

jest.mock('../reporters/default_reporter');
jest.mock('node-notifier', () => ({
  notify: jest.fn(),
}));

const initialContext: TestSchedulerContext = {
  firstRun: true,
  previousSuccess: false,
};

const aggregatedResultsSuccess: AggregatedResult = {
  numFailedTestSuites: 0,
  numFailedTests: 0,
  numPassedTestSuites: 1,
  numPassedTests: 3,
  numRuntimeErrorTestSuites: 0,
  numTotalTestSuites: 1,
  numTotalTests: 3,
  success: true,
};

const iconDir = path => {
  if (path.endsWith(SUCCESS_ICON_PATH)) {
    return 'SUCCESS_ICON';
  } else if (path.endsWith(FAILURE_ICON_PATH)) {
    return 'FAILURE_ICON';
  } else {
    return 'ICON_NOT_FOUND';
  }
};

test('.addReporter() .removeReporter()', () => {
  const scheduler = new TestScheduler({}, {}, initialContext);
  const reporter = new NotifyReporter();
  scheduler.addReporter(reporter);
  expect(scheduler._dispatcher._reporters).toContain(reporter);
  scheduler.removeReporter(NotifyReporter);
  expect(scheduler._dispatcher._reporters).not.toContain(reporter);
});

test('test always', () => {
  const notify = require('node-notifier');

  const reporter = new NotifyReporter(
    {notify: true, notifyMode: 'always'},
    {},
    initialContext,
  );
  reporter.onRunComplete(new Set(), aggregatedResultsSuccess);
  expect(
    notify.notify.mock.calls.map(([{icon, message, title}]) => ({
      icon: iconDir(icon),
      message,
      title,
    })),
  ).toMatchSnapshot();
});
