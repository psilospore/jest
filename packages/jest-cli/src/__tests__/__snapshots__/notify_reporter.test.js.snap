// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test always 1`] = `
Array [
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
  Object {
    "icon": true,
    "message": "⛔️ 3 of 3 tests failed",
    "title": "100% Failed",
  },
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
  Object {
    "icon": true,
    "message": "⛔️ 3 of 3 tests failed",
    "title": "100% Failed",
  },
  Object {
    "icon": true,
    "message": "⛔️ 3 of 3 tests failed",
    "title": "100% Failed",
  },
]
`;

exports[`test change 1`] = `
Array [
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
  Object {
    "icon": true,
    "message": "⛔️ 3 of 3 tests failed",
    "title": "100% Failed",
  },
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
  Object {
    "icon": true,
    "message": "⛔️ 3 of 3 tests failed",
    "title": "100% Failed",
  },
]
`;

exports[`test failure-change 1`] = `
Array [
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
  Object {
    "icon": true,
    "message": "⛔️ 3 of 3 tests failed",
    "title": "100% Failed",
  },
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
  Object {
    "icon": true,
    "message": "⛔️ 3 of 3 tests failed",
    "title": "100% Failed",
  },
  Object {
    "icon": true,
    "message": "⛔️ 3 of 3 tests failed",
    "title": "100% Failed",
  },
]
`;

exports[`test success 1`] = `
Array [
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
]
`;

exports[`test success-change 1`] = `
Array [
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
  Object {
    "icon": true,
    "message": "⛔️ 3 of 3 tests failed",
    "title": "100% Failed",
  },
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
  Object {
    "icon": true,
    "message": "✅ 3 tests passed",
    "title": "100% Passed",
  },
  Object {
    "icon": true,
    "message": "⛔️ 3 of 3 tests failed",
    "title": "100% Failed",
  },
]
`;
